sensor/common/virtualmachine/component.go

```diff
@@ -0,0 +1,27 @@
+package virtualmachine
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common"
+)
+
+// Component provides functionality to send virtual machines to Central.
+type Component interface {
+	common.SensorComponent
+
+	Send(ctx context.Context, vm *storage.VirtualMachine) error
+}
+
+// NewComponent returns the virtual machine component for Sensor to use.
+func NewComponent() Component {
+	return &componentImpl{
+		centralReady:    concurrency.NewSignal(),
+		lock:            &sync.Mutex{},
+		stopper:         concurrency.NewStopper(),
+		virtualMachines: make(chan *storage.VirtualMachine),
```

Comment:
We could consider the writer of this channel as whatever calls the `Send` function (in this case the service). I'd probably create the channel in the service and pass it as a input parameter to the component.

Reply:
I don't understand why this would be beneficial. I chose to let the service communicate with the handler via `Send` (instead of a channel like in compliance) because it is simpler to return status codes for the unary virtual machine gRPC requests. I would then consider `virtualMachines` an implementation detail of the handler, which it uses to synchronize its internal go routines.